src/overlays/actors/ovl_En_Door/z_en_door.c

```diff
@@ -27,29 +39,328 @@ const ActorInit En_Door_InitVars = {
+        this->actor.posRot.pos.z += zOffset;
+    }
+    Actor_SetHeight(&this->actor, 70.0f);
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/EnDoor_Init.s")
+#endif
+
+void EnDoor_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    TransitionActorEntry* transitionEntry;
+    EnDoor* this = THIS;
+
+    transitionEntry = &globalCtx->transitionActorList[(u16)this->actor.params >> 0xA];
+    if (transitionEntry->id < 0) {
+        transitionEntry->id = -transitionEntry->id;
+    }
+}
+
+void EnDoor_SetupType(EnDoor* this, GlobalContext* globalCtx) {
+    s32 doorType;
+
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->currentObject)) {
+        doorType = (this->actor.params >> 7) & 7;
+        this->actor.flags &= ~0x10;
+        this->actor.objBankIndex = this->currentObject;
+        this->actionFunc = EnDoor_Idle;
+        if (doorType == DOOR_TIME_LOCKED) {
+            doorType = (gSaveContext.dayTime > 0xC000 && gSaveContext.dayTime < 0xE000) ? DOOR_NOLOAD : DOOR_TALKING;
+        }
+        this->actor.posRot.rot.y = 0x0000;
+        if (doorType == DOOR_LOCKED) {
+            if (!Flags_GetSwitch(globalCtx, this->actor.params & 0x3F)) {
+                this->lockTimer = 10;
+            }
+        } else if (doorType == DOOR_AJAR) {
+            if (func_8002DB8C(&this->actor, &PLAYER->actor) > DOOR_SLAM_RANGE) {
+                this->actionFunc = EnDoor_CheckRange;
+                this->actor.posRot.rot.y = -0x1800;
+            }
+        } else if (doorType == DOOR_TALKING) {
+            this->actor.textId = (this->actor.params & 0x3F) + 0x0200;
+            if (this->actor.textId == 0x0229 && !(gSaveContext.eventChkInf[1] & 0x10)) {
+                // Talon's house door. If Talon has not been woken up at Hyrule Castle
+                // this door should be openable at any time of day. Note that there is no
+                // check for time of day as the scene setup for Lon Lon merely initializes
+                // the door with a different text id between day and night setups
+                doorType = DOOR_NOLOAD;
+            } else {
+                this->actionFunc = EnDoor_WaitForCheck;
+                this->actor.flags |= 0x8000009;
+            }
+        }
+        // Replace the door type it was loaded with by the new type
+        this->actor.params = (this->actor.params & ~0x380) | (doorType << 7);
+    }
+}
+
+#ifdef NON_MATCHING
+// Regalloc only
+void EnDoor_Idle(EnDoor* this, GlobalContext *globalCtx) {
+    Player* player = PLAYER;
+    s32 doorType;
+    Vec3f sp2C;
+    s8 numKeys;
+    s16 phi_v0;
+    s16 temp_v0;
+
+    doorType = (this->actor.params >> 7) & 7;
+    func_8002DBD0(&this->actor, &sp2C, &player->actor.posRot.pos);
+    if (this->unk_191 != 0) {
+        this->actionFunc = EnDoor_Open;
+        SkelAnime_ChangeAnimPlaybackStop(&this->skelAnime, D_809FCECC[this->unk_190], (((s32)player->stateFlags1 * 0x10) < 0) ? 0.75f : 1.5f);
```

Comment:
```suggestion
        SkelAnime_ChangeAnimPlaybackStop(&this->skelAnime, D_809FCECC[this->unk_190], (player->stateFlags1 & 0x8000000) ? 0.75f : 1.5f);
```
maybe

Reply:
This worked but didn't affect matching unfortunately, but it's definitely a lot cleaner.